sense-android-library/test/java/nl/sense_os/datastorageengine/CommonSenseProxyTest.java

```diff
@@ -0,0 +1,560 @@
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * {@link CommonSenseProxyTest} tests whether {@link CommonSenseProxy} covers the full CommonSense API
+ */
+@RunWith(RobolectricTestRunner.class)
+@Config(constants = BuildConfig.class, sdk = 21)
+public class CommonSenseProxyTest {
+
+    Map<String, String> newUser;
+    CommonSenseProxy proxy = new CommonSenseProxy(false, CSUtils.APP_KEY);
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+
+    @Before
+    public  void setUp () throws IOException{
+        newUser = CSUtils.createCSAccount();
+
+    }
+
+    @Test
+    public void testLoginUserWithValidUsernameAndValidPassword() throws IOException, JSONException{
+        String session_id = proxy.loginUser(newUser.get("username"),newUser.get("password"));
+        assertNotNull("session_id returned from server is null", session_id);
+        assertFalse("session_id returned from server is empty",session_id.isEmpty());
+    }
+    @Test
+    public void testLoginUserWithValidUsernameAndWrongPassword() throws IOException, JSONException{
+        thrown.expect(IOException.class);
+        thrown.expectMessage("could not get InputStream");
+        proxy.loginUser(newUser.get("username"), "123456789");
+    }
+    @Test
+    public void testLoginUserWithValidUsernameAndNullPassword() throws IOException, JSONException{
+        thrown.expect(IOException.class);
+        thrown.expectMessage("invalid input of username or password");
+        proxy.loginUser(newUser.get("username"),null);
+    }
+    @Test
+    public void testLoginUserWithValidUsernameAndEmptyPassword() throws IOException, JSONException{
+        thrown.expect(IOException.class);
+        thrown.expectMessage("invalid input of username or password");
+        proxy.loginUser(newUser.get("username"),"");
+    }
+    @Test
+    public void testLoginUserWithNullUsernameAndValidPassword() throws IOException, JSONException{
+        thrown.expect(IOException.class);
+        thrown.expectMessage("invalid input of username or password");
+        proxy.loginUser(null,newUser.get("password"));
+    }
+    @Test
+    public void testLoginUserWithEmptyUsernameAndValidPassword() throws IOException, JSONException{
+        thrown.expect(IOException.class);
+        thrown.expectMessage("invalid input of username or password");
+        proxy.loginUser("",newUser.get("password"));
+    }
+    @Test
+    public void testLogoutCurrentUserWithValidSessionID() throws IOException, JSONException{
+        // log in first in order to log out
+        String session_id = proxy.loginUser(newUser.get("username"), newUser.get("password"));
+        boolean result = proxy.logoutCurrentUser(session_id);
+        assertTrue("current user cannot be successfully logged out", result);
```

Comment:
@excaliburnan this does not look like the proper response, it should succeed right?